src/backend/opencl/kernel/scan_first.cl

```diff
@@ -118,11 +118,12 @@ void bcast_first_kernel(__global To *oData, KParam oInfo,
 
             To accum = tData[groupId_x - 1];
 
+            int offset = !inclusive_scan;
             for (int k = 0, id = xid;
-                 k < lim && id < oInfo.dims[0];
+                 k < lim && id + offset < oInfo.dims[0];
```

Comment:
You could just increment id when initializing the loop instead of every time its used.